fn_ioc_parser/fn_ioc_parser/components/function_ioc_parser.py

```diff
@@ -0,0 +1,99 @@
+            log.info("ioc_parser_artifact_id: %s", ioc_parser_artifact_id)
+            log.info("ioc_parser_artifact_type: %s", ioc_parser_artifact_type)
+            log.info("ioc_parser_artifact_value: %s", ioc_parser_artifact_value)
+
+            yield StatusMessage("Extracting IOCs from given Artifact/attachment data")
+
+            # Initialising the Resilient Rest client for making api calls to Resilient system
+            resilient_client = self.rest_client()
+
+            # Initialising the IOC Parser Helper class
+            IOCHelp_obj = IOCParserHelper()
+
+            if ioc_parser_artifact_id:
+                # A block to parse and download the data from Artifacts
+                if ioc_parser_artifact_type.lower().strip() in ['string', 'email subject', 'email body']:
+                    ioc_parser_data = ioc_parser_artifact_value
+                else:
+                    metadata_uri = IOCHelp_obj.ARTIFACT_META_DATA_URL.format(ioc_parser_incident_id, ioc_parser_artifact_id)
+                    data_uri = IOCHelp_obj.ARTIFACT_DATA_URL.format(ioc_parser_incident_id, ioc_parser_artifact_id)
+
+                    # Getting the Meta data from the Resilient for an artifact
+                    metadata_data = resilient_client.get(metadata_uri)
+                    if metadata_data.get('attachment'):
+                        # Getting the file content from Resilient system
+                        attachment_data = resilient_client.get_content(data_uri)
+
+                        attachment_file_name = metadata_data.get('attachment').get('name')
+                        ioc_parser_data = IOCParserHelper.extract_text_from_bytes_data(attachment_file_name, attachment_data)
+                    else:
+                        ioc_parser_data = ioc_parser_artifact_value
+            else:
+                # A block to parse and download the data from attachments
+                metadata_uri = IOCHelp_obj.ATTACHMENT_META_DATA_URL.format(ioc_parser_incident_id, ioc_parser_attachment_id)
+                data_uri = IOCHelp_obj.ATTACHMENT_DATA_URL.format(ioc_parser_incident_id, ioc_parser_attachment_id)
+                if ioc_parser_task_id:
+                    metadata_uri = IOCHelp_obj.TASK_META_DATA_URL.format(ioc_parser_task_id, ioc_parser_attachment_id)
+                    data_uri = IOCHelp_obj.TASK_DATA_URL.format(ioc_parser_task_id, ioc_parser_attachment_id)
+
+                # Getting the Meta data from the Resilient for an artifact
+                metadata_data = resilient_client.get(metadata_uri)
+
+                # Getting the file content from Resilient system
+                attachment_data = resilient_client.get_content(data_uri)
+
+                attachment_file_name = metadata_data.get('name')
+                ioc_parser_data = IOCParserHelper.extract_text_from_bytes_data(attachment_file_name, attachment_data)
+
+            ioc_text_obj = IOCParser(ioc_parser_data)
```

Comment:
* Should this be surrounded by a `try/except` block?
* Will it ever fail and `ioc_text_obj` be 'None`?
* Should we handle this case?

Reply:
we don't needs try/except block here because it never fails. `ioc_text_obj` will be always having class instance object.

Reply:
OK